dvc/checkout.py

```diff
@@ -176,7 +177,11 @@ def _checkout_file(
 
 
 def _remove_redundant_files(path_info, fs, obj, cache, force):
-    existing_files = set(fs.walk_files(path_info))
+    if fs.scheme == Schemes.LOCAL and fs.repo:
+        walk_files = fs.repo.dvcignore(fs.walk(path_info), walk_files=True)
+    else:
+        walk_files = fs.walk_files(path_info)
+    existing_files = set(walk_files)
```

Comment:
`walk_files` should probably be `files` to avoid confusion with the function.

How about:

```suggestion
    files = set(cache.repo.dvcignore(fs, walk_files=True))
```

unlike `fs.repo`, `cache.repo` is more persistent (though temporarily) so we can rely on it more.

Or maybe even:
```
    files = set(cache.repo.dvcignore.walk_files(fs, path_info))
```

so that dvcignore has `walk` and `walk_files` methods.

Also, for the record, I'm refactoring this right now to decouble repo from fs, and `fs.repo` will go away soon, as well as `cache.repo`. We'll likely start passing `dvcignore`(as well as state) to checkout as kwargs. No action needed in this PR, just FYI.

Reply:
How about the `RepoFileSystem`? Seems that it's difficult to be decoupled from `repo`.

Reply:
In my opinion, maybe we should pass a list of individual files instead of path and file systems other things in the last three cases.

Reply:
> In my opinion, maybe we should pass a list of individual files instead of path and file systems other things in the last three cases.

Could you elaborate, please?